### PR TITLE
Missing test cases for MODPERMS-36

### PR DESCRIPTION
**Pull request by adamdickmeiss**
This issue was filed a long time ago and was fixed .. this adds a few tests, so we can close the issue

### CODE REVIEW

File: src/test/java/org/folio/permstest/RestVerticleTest.java [hunk 1]
```diff
@@ -189,8 +191,42 @@ public void testPermsSeq(TestContext context) {
...
- public void testGroup(TestContext context){
+  @Test
+  public void testPermsUsersPermissionsInvalidUUID(TestContext context)
+      throws InterruptedException, ExecutionException, TimeoutException {
+    String url = "http://localhost:" + port + "/perms/users/123/permissions";
+
+    CompletableFuture<Response> response = new CompletableFuture();
+    send(url, context, HttpMethod.POST, postPermUserPermRequest,
+        SUPPORTED_CONTENT_TYPE_JSON_DEF, 400,  new HTTPResponseHandler(response));
+    Response addPermsResponse = response.get(5, TimeUnit.SECONDS);
```
Comment (hjiebsco) [inline src/test/java/org/folio/permstest/RestVerticleTest.java:202]: Consider to add this after line 202
`assertEquals(400, addPermsResponse.code);`

File: src/test/java/org/folio/permstest/RestVerticleTest.java [hunk 2]
```diff
@@ -189,8 +191,42 @@ public void testPermsSeq(TestContext context) {
...
+
+  @Test
+  public void testPermsUsersPutInvalidUUID(TestContext context)
+      throws InterruptedException, ExecutionException, TimeoutException {
+    String url = "http://localhost:" + port + "/perms/users/123";
+
+    CompletableFuture<Response> response = new CompletableFuture();
+    send(url, context, HttpMethod.PUT, postPermUsersRequest,
+        SUPPORTED_CONTENT_TYPE_JSON_DEF, 400,  new HTTPResponseHandler(response));
+    Response addPermsResponse = response.get(5, TimeUnit.SECONDS);
```
Comment (hjiebsco) [inline src/test/java/org/folio/permstest/RestVerticleTest.java:213]: Consider to add this after line 213. Also consider to rename `addPermsResponse` to be PUT related
`assertEquals(404, addPermsResponse.code);`

File: src/test/java/org/folio/permstest/RestVerticleTest.java [hunk 3]
```diff
@@ -189,8 +191,42 @@ public void testPermsSeq(TestContext context) {
...
+
+  @Test
+  public void testPermsUsersDeleteInvalidUUID(TestContext context)
+      throws InterruptedException, ExecutionException, TimeoutException {
+    String url = "http://localhost:" + port + "/perms/users/123";
+
+    CompletableFuture<Response> response = new CompletableFuture();
+    send(url, context, HttpMethod.DELETE, "",
+        SUPPORTED_CONTENT_TYPE_JSON_DEF, 400,  new HTTPResponseHandler(response));
+    Response addPermsResponse = response.get(5, TimeUnit.SECONDS);
```
Comment (hjiebsco) [inline src/test/java/org/folio/permstest/RestVerticleTest.java:224]: Consider to add this after line 224. Also consider to rename `addPermsResponse` to be DELETE related
`assertEquals(404, addPermsResponse.code);`

File: src/test/java/org/folio/permstest/RestVerticleTest.java (hunk 1 above)
Comment (adamdickmeiss) [reply src/test/java/org/folio/permstest/RestVerticleTest.java:202]: OMG.. send does not use the errorCode parameter. I'll fix that..

File: src/main/java/org/folio/rest/impl/PermsAPI.java [hunk 4]
```diff
@@ -140,1165 +140,1094 @@ private static void report(String noise, Logger logger) {
...
-      });
+      }
     } catch (Exception e) {
-      String errStr = "Error running vertx on context:" + e.getLocalizedMessage();
+      String errStr = "Error running vertx on context:" + e.getMessage();
       logger.error(errStr, e);
-      asyncResultHandler.handle(io.vertx.core.Future.succeededFuture(
-        GetPermsUsersResponse.respond500WithTextPlain(getErrorResponse(errStr))));
+      asyncResultHandler.handle(Future.succeededFuture(
+          GetPermsUsersResponse.respond500WithTextPlain(errStr)));
```
Comment (hjiebsco) [inline src/main/java/org/folio/rest/impl/PermsAPI.java:195]: Seems that `getErrorResponse()` was removed after the refactoring.

File: src/main/java/org/folio/rest/impl/PermsCache.java [hunk 5]
```diff
@@ -59,15 +58,15 @@ private PermsCache() {
...
       if (ar.succeeded()) {
-        future.complete(ar.result().expandPerms(perms));
+        promise.complete(ar.result().expandPerms(perms));
       } else {
-        future.fail(ar.cause());
+        promise.fail(ar.cause());
       }
     });
-    return future;
+    return promise.future();
```
Comment (julianladisch) [inline src/main/java/org/folio/rest/impl/PermsCache.java:69]: This should be replaced by map, no need to manually handle the fail case:

```
return getPermCache(vertxContext, tenantId, new HashSet<>(perms)).map(permCache -> permCache.expandPerms(perms));
```

File: src/main/java/org/folio/rest/impl/PermsCache.java [hunk 6]
```diff
@@ -79,15 +78,15 @@ private PermsCache() {
...
       if (ar.succeeded()) {
-        future.complete(ar.result().getFullPerm(permissionName));
+        promise.complete(ar.result().getFullPerm(permissionName));
       } else {
-        future.fail(ar.cause());
+        promise.fail(ar.cause());
       }
     });
-    return future;
+    return promise.future();
```
Comment (julianladisch) [inline src/main/java/org/folio/rest/impl/PermsCache.java:89]: Should be replaced by map as above.

**Pull request merged**